pkg/caret/R/preProcess.R

```diff
@@ -236,9 +236,11 @@ preProcess.default <- function(x, method = c("center", "scale"),
   ## check for zero-variance predictors
   if(any(names(method) == "zv")){
     if(is.data.frame(x)) {
-      is_zv <- unlist(lapply(x[, !(colnames(x) %in% method$ignore), drop = FALSE], function(x) length(unique(x)) <= 1))
```

Comment:
Please represent the true brach here as a logical and not integer (e.g. `length(na.omit(unique(x)))`)